### PR TITLE
Fix count() handling in ScalarAggregationToJoinRewriter

### PR DESCRIPTION
**Pull request by kasiafi**
Fixes a bug in decorrelating a scalar subquery with count() aggregation.

Before this change, the aggregation function `count(argument)` was rewritten in the way that it returned a row count instead of non-null value count for the `argument`.

### CODE REVIEW

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -3639,6 +3639,18 @@ public void testCorrelatedScalarSubqueriesWithScalarAggregation()
...
+        // count rows
+        assertQuery("SELECT (SELECT count(*) FROM (VALUES (1, true), (null, true)) t2(a, b) WHERE t2.b = t1.b) FROM (VALUES (true)) t1(b)",
+                "VALUES (2)");
+
+        // count rows
+        assertQuery("SELECT (SELECT count() FROM (VALUES (1, true), (null, true)) t2(a, b) WHERE t2.b = t1.b) FROM (VALUES (true)) t1(b)",
+                "VALUES (2)");
+
+        // count non null values
+        assertQuery("SELECT (SELECT count(a) FROM (VALUES (1, true), (null, true)) t2(a, b) WHERE t2.b = t1.b) FROM (VALUES (true)) t1(b)",
```
Comment (kokosing) [inline presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: - can you use use `inner_table` for `t2` and `outer_table` for `t1`?

File: presto-main/src/main/java/io/prestosql/sql/planner/optimizations/ScalarAggregationToJoinRewriter.java [hunk 2]
```diff
@@ -174,7 +174,8 @@ private PlanNode rewriteScalarAggregation(
         for (Map.Entry<Symbol, Aggregation> entry : scalarAggregation.getAggregations().entrySet()) {
             Aggregation aggregation = entry.getValue();
             Symbol symbol = entry.getKey();
-            if (aggregation.getSignature().getName().equals("count")) {
+            // Only count() and count(*) require rewriting to count(non_null) in order to preserve the row count. count(argument) shouldn't be rewritten.
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/sql/planner/optimizations/ScalarAggregationToJoinRewriter.java]: nice catch!

**Comment by kasiafi** [comment]
Applied.

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**